packages/widget-chat/src/components/activity-list/index.js

```diff
@@ -1,39 +1,78 @@
-      lastActorId = activity.actor.id;
-      lastVerb = activity.verb;
-      const isFlagged = flags && flags.some((flag) => flag.activityUrl === activity.url);
-      return (
-        <ActivityItem
-          activity={activity.object}
-          avatarUrl={avatars[activity.actor.id]}
-          id={activity.id}
-          isAdditional={additional}
-          isFlagged={isFlagged}
-          isSelf={currentUserId === activity.actor.id}
-          key={activity.id}
-          name={activity.actor.displayName}
-          onActivityDelete={onActivityDelete}
-          onActivityFlag={onActivityFlag}
-          timestamp={formatDate(activity.published)}
-          verb={activity.verb}
+  let lastActorId, lastDay, lastVerb;
+  const now = moment();
+  const {avatars, currentUserId, flags, lastAcknowledgedActivity, onActivityDelete, onActivityFlag} = props;
+  const items = [];
+  let shouldDisplayNewMessageMarker = false;
+  props.activities.forEach((activity) => {
+    // Insert day separator if this activity and last one happen on a different day
+    const activityMoment = moment(activity.published, moment.ISO_8601);
+    const activityDay = activityMoment.endOf(`day`);
+    const sameDay = activityDay.diff(lastDay, `days`) === 0;
+    if (lastDay && !sameDay) {
+      items.push(
+        <DaySeparator
+          fromDate={lastDay}
+          key={`day-separtor-${activity.id}`}
+          now={now}
+          toDate={activityDay}
+        />
+      );
+    }
+    lastDay = activityDay;
+
+    if (shouldDisplayNewMessageMarker) {
+      const newMessages = ( // eslint-disable-line no-extra-parens
+        <FormattedMessage
+          defaultMessage={`NEW MESSAGES`}
+          description={`Indicator to display that new, unread messages follow`}
+          id={`newMessages`}
         />
       );
-    });
+      items.push(<ListSeparator classNames={[`upper`, `blue`]} primaryText={newMessages} />);
+      shouldDisplayNewMessageMarker = false;
+    }
+
+    // additional items don't repeat user avatar and name
+    const additional = sameDay && lastActorId === activity.actor.id && lastVerb === activity.verb;
+    lastActorId = activity.actor.id;
+    lastVerb = activity.verb;
+    const isFlagged = flags && flags.some((flag) => flag.activityUrl === activity.url);
+    items.push(
+      <ActivityItem
+        activity={activity.object}
+        avatarUrl={avatars[activity.actor.id]}
+        id={activity.id}
+        isAdditional={additional}
+        isFlagged={isFlagged}
+        isSelf={currentUserId === activity.actor.id}
+        key={activity.id}
+        name={activity.actor.displayName}
+        onActivityDelete={onActivityDelete}
+        onActivityFlag={onActivityFlag}
+        timestamp={formatDate(activity.published)}
+        verb={activity.verb}
+      />
+    );
+
+    // if this activity is last acked by user render the banner
+    if (lastAcknowledgedActivity && lastAcknowledgedActivity === activity.id) {
```

Comment:
nitpick, may want to rename `lastAcknowledgeActivity` to `lastAcknowledgeActivityId`